sklearn/metrics/_classification.py

```diff
@@ -2979,32 +2981,36 @@ def log_loss(y_true, y_pred, *, normalize=True, sample_weight=None, labels=None)
                 "got {0}.".format(lb.classes_)
             )
 
-    transformed_labels = lb.transform(y_true)
+    xp_float = _find_matching_floating_dtype(y_true, y_pred, xp=xp)
+    transformed_labels = xp.asarray(
+        lb.transform(_convert_to_numpy(y_true, xp=xp)), dtype=xp_float, device=device_
+    )
```

Comment:
Can we change the naming of the `xp_float` variable into something more intuitive? Maybe like this?
```suggestion
    floating_dtype = _find_matching_floating_dtype(y_true, y_pred, xp=xp)
    transformed_labels = xp.asarray(
        lb.transform(_convert_to_numpy(y_true, xp=xp)), dtype=floating_dtype, device=device_
    )
```